file-formats/smtg/type/zob_aff_smtg_v1.intf.abap

```diff
@@ -0,0 +1,156 @@
+      "! Created by
+      "! $showAlways
+      created_by      type string,
+      "! <p class="shorttext">Last Changed By</p>
+      "! last changed by
+      "! $showAlways
+      last_changed_by type string,
+    end of ty_administrative_data.
+
+  types:
+    "! <p class="shorttext">Header</p>
+    "! Contains identifying and descriptive information about the template.
+    begin of ty_template_header,
+      "! <p class="shorttext">Template Name</p>
+      "! Unique identifier for the template
+      "! $required
+      template_id  type smtg_tmpl_name,
+      "! <p class="shorttext">CDS View</p>
+      "! The corresponding CDS view name for the template
+      "! $required
+      cds_view     type smtg_cds_view,
+      "! <p class="shorttext">Description</p>
+      "! Brief description of the template
+      description  type smtg_tmpl_description,
+      "! <p class="shorttext">Predelivered</p>
+      "! Indicator if the template is pre-delivered
+      predelivered type abap_bool,
+    end of ty_template_header.
+
+  types:
+    "! <p class="shorttext">General Information</p>
+    "! Combines template header and administrative data.
+    begin of ty_general_information,
+      "! <p class="shorttext">Template Header</p>
+      "! Contains basic template identifiers and description
+      "! $required
+      template_header     type ty_template_header,
+      "! <p class="shorttext">Administrative Data</p>
+      "! Contains info about template creation and modification
+      "! $required
+      administrative_data type ty_administrative_data,
+    end of ty_general_information.
+
+
+  " Email Template Body --------------------------------------------------------------------------------------------
+
+  types:
+    "! <p class="shorttext">Template Content</p>
+    "! Holds the actual content of the email template.
+    begin of ty_template_content,
+      "! <p class="shorttext">Body HTML</p>
+      "! The HTML body content of the email template
+      body_html       type string,
+      "! <p class="shorttext">Body Plain Text</p>
+      "! The plain text body content of the email template
+      body_plain_text type string,
+    end of ty_template_content.
+
+
+  " Email Template body options --------------------------------------------------------------------------------------------
+
+  types:
+    "! <p class="shorttext">Template Language</p>
+    "! Language options for the template
+    begin of ty_template_language,
+      "! <p class="shorttext">Language</p>
+      "! Language code
+      "! $required
+      country type string,
+      "! <p class="shorttext">HTML</p>
+      "! Indicates if HTML format is available
+      html    type abap_bool,
+      "! <p class="shorttext">Plain</p>
+      "! Indicates if plain text is available
+      Plain   type abap_bool,
+      "! <p class="shorttext">Auto</p>
+      "! Indicates if automatic plain text update is enabled
+      Auto    type abap_bool,
```

Comment:
```suggestion
      auto    type abap_bool,
```

Reply:
When does the translation from html to plain text happen? When the html is updated, when the email is sent...? Could you clarify what the "auto" flag does? Is the "auto" flag language dependent?
 
The language dependent fields/texts (html and plain text) will be in separate files and not in the json file of the main object.

Reply:
For documentation purposes: The auto flag is not language dependent and cannot be changed by translators. If the flag is set to true, the conversion of html to plain text happens on save in the current gui editor.